torch/csrc/aot_inductor/aten_shim_common.cpp

```diff
@@ -0,0 +1,492 @@
+  AOT_INDUCTOR_CHECK(
+      kAOTInductorBool == int8_t(c10::ScalarType::Bool),
+      "kAOTInductorBool != c10::ScalarType::Bool")
+  AOT_INDUCTOR_CHECK(
+      kAOTInductorQInt8 == int8_t(c10::ScalarType::QInt8),
+      "kAOTInductorQInt8 != c10::ScalarType::QInt8")
+  AOT_INDUCTOR_CHECK(
+      kAOTInductorQUInt8 == int8_t(c10::ScalarType::QUInt8),
+      "kAOTInductorQUInt8 != c10::ScalarType::QUInt8")
+  AOT_INDUCTOR_CHECK(
+      kAOTInductorQInt32 == int8_t(c10::ScalarType::QInt32),
+      "kAOTInductorQInt32 != c10::ScalarType::QInt32")
+  AOT_INDUCTOR_CHECK(
+      kAOTInductorBFloat16 == int8_t(c10::ScalarType::BFloat16),
+      "kAOTInductorBFloat16 != c10::ScalarType::BFloat16")
+
+#ifndef NDEBUG
+  live_buffer_ = new std::unordered_set<void*>;
+#endif
+}
+
+void aot_inductor_destroy() {
+#ifndef NDEBUG
+  AOT_INDUCTOR_CHECK(
+      live_buffer_->size() == 0, "Live intermediate buffer leaking");
+  delete live_buffer_;
+#endif
+}
+
+void aot_inductor_free(AOTInductorTensorHandle aot_tensor) {
+  AOTInductorTensor* t = static_cast<AOTInductorTensor*>(aot_tensor);
+  AOT_INDUCTOR_CHECK(t != nullptr, "Invalid AOTInductorTensorHandle")
+  AOT_INDUCTOR_CHECK(
+      t->deleter != nullptr, "Invalid AOTInductorTensorHandle deleter")
+
+  if (t->deleter == deleteNothing)
+    return;
+
+  DeleterFnPtr deleter = DeleterFnPtr(t->deleter);
+#ifndef NDEBUG
+  live_buffer_->erase(t->data_ptr);
+#endif
+  deleter(t->data_ptr);
+}
+
+void* aot_inductor_data_ptr(AOTInductorTensorHandle aot_tensor) {
+  // t->storage_offset is in the number of elements, need to turn it
+  // into bytes
+  AOTInductorTensor* t = static_cast<AOTInductorTensor*>(aot_tensor);
+  char* base = static_cast<char*>(t->data_ptr);
+  size_t offset_in_bytes = t->storage_offset * get_dtype_bytes(t->dtype);
+  return (void*)(base + offset_in_bytes);
+}
+
+AOTInductorTensorHandle convert_input_output_to_aot_tensor(
+    AtenTensorHandle aten_tensor) {
+  // Input and output tensors are managed by the external runtime, so they are
+  // kept live in the AOTInductor runtime, and thus we set their corresponding
+  // deleter to deleteNothing.
+  at::Tensor* t = static_cast<at::Tensor*>(aten_tensor);
+  return register_tensor(
+      t->data_ptr(),
+      (void*)deleteNothing,
+      t->dim(),
+      t->sizes().data(), // sizes will stay live, so no need to register them
+      t->strides()
+          .data(), // strides will stay live, so no need to register them
+      t->storage_offset(),
+      static_cast<AOTInductorDeviceType>(t->device().type()),
+      static_cast<AOTInductorDeviceIndex>(t->device().index()),
+      static_cast<AOTInductorScalarType>(t->scalar_type()));
+}
+
+AOTInductorTensorHandle aot_inductor_empty_strided(
+    int64_t ndim,
+    const int64_t* sizes_ptr,
+    const int64_t* strides_ptr,
+    AOTInductorDeviceType device_type,
+    AOTInductorDeviceIndex device_index,
+    AOTInductorScalarType dtype) {
```

Comment:
Assume that device guard is set appropriately (who is setting the device guard? modifying the cuda runtime state, which is shared)

`aoti_runtime_cuda_alloc(int64_t nbytes)`